cdap-docs/admin-manual/source/operations/logging.rst

```diff
@@ -50,28 +102,409 @@ CDAP supports logging through standard `SLF4J (Simple Logging Facade for Java)
+    <appendix-cdap-default-logging>` file as described in 
+    :ref:`logging-monitoring-logging-pipeline-configuration`.
+
+  - Custom pipelines can be configured by adding an XML file in a prescribed location.
+    Each pipeline buffers log messages in-memory and sorts them based on their timestamp. 
+   
+  - Metrics are created based on the results of the pipelines, and passed to the Metrics
+    Service.
+   
+  - For security, the files written out to persistent storage in HDFS have permissions set
+    such that they are accessible only by the ``cdap`` user.
+
+  - Log.saver also emits metrics for each *info, debug,* or *error* message received. This
+    helps in monitoring the amount of logs emitted by applications, application's programs,
+    system services, etc. and can be retrieved by querying the :ref:`CDAP metrics system
+    <http-restful-api-metrics>`.
+
+.. figure:: /_images/logging/logging-framework.png
+    :figwidth: 100%
+    :width: 800px
+    :align: center
+
+    **CDAP Logging Framework:** From YARN containers, through Kafka and the Log Saver
+    Service, to HDFS
+
+Logging Standards
+-----------------
+Logging uses the standard `SLF4J (Simple Logging Facade for Java)
+<http://www.slf4j.org/manual.html>`__ APIs and `Logback
+<https://logback.qos.ch/manual>`__. Logging is configured using instances of Logback's
+"logback" file, consisting of *logging pipelines* with *logging appenders*:
+
+  - A **logging pipeline** is a process that consumes log events from Kafka, invokes
+    *logback appenders* defined in its configuration, buffering, sorting, and then
+    persisting the log messages.
+
+  - A **logback appender** (or *appender*) is a Java class, responsible for writing the
+    log events to persistent storage and maintaining metadata about that storage.
+
+Configuration Files
+-------------------
+- The ``logback-container.xml`` configures the logs of YARN containers, both locally and to Kafka,
+  for application programs and any :ref:`CDAP System Services
+  <logging-monitoring-system-services-logs>` (such as the *Metrics Service*) that run in YARN. 
+- The ``logback.xml`` configures the edge node-located :ref:`CDAP System Services
+  <logging-monitoring-system-services-logs>`, such as the *App Fabric* or *Data Service*.
+- Both of these files are, in Distributed CDAP, located in ``/etc/cdap/conf``.
+- In the case of :ref:`Standalone CDAP <modes-data-application-platform>`, as containers
+  are replaced with Java threads, only a single file (``logback.xml``, located in
+  ``<cdap-sdk-home>/conf``) is used, replacing the ``logback.xml`` and
+  ``logback-container.xml`` of Distributed CDAP.
+- The ``cdap-site.xml`` controls the CDAP Log Pipeline for CDAP System Services, and the
+  reading, buffering, and memory allocated.
+- Custom "logback" files, if present in a specified directory, trigger the creation of
+  custom logging pipelines, with a pipeline created for each custom file.
```

Comment:
This whole paragraph repeats information from further above. is that on purpose?